File: serialization/src/test/java/net/corda/serialization/internal/amqp/testutils/TestSerializationContext.java
package net.corda.serialization.internal.amqp.testutils;

import net.corda.core.serialization.AMQPSerializationContext;
import net.corda.serialization.internal.AMQPSerializationContextImpl;
import net.corda.serialization.internal.AllWhitelist;


import java.util.HashMap;
import java.util.Map;

public class TestSerializationContext {

    private static Map<Object, Object> serializationProperties = new HashMap<>();

    public static AMQPSerializationContext testSerializationContext = new AMQPSerializationContextImpl(
        ClassLoader.getSystemClassLoader(),
        AllWhitelist.INSTANCE,
        serializationProperties,
        false,
        AMQPSerializationContext.UseCase.Testing,
        null);
}

File: serialization/src/test/kotlin/net/corda/serialization/internal/amqp/testutils/TestSerializationContext.kt
package net.corda.serialization.internal.amqp.testutils

import net.corda.core.serialization.AMQPSerializationContext
import net.corda.core.serialization.SerializationContext
import net.corda.serialization.internal.AMQPSerializationContextImpl
import net.corda.serialization.internal.AllWhitelist
import net.corda.serialization.internal.SerializationContextImpl
import net.corda.serialization.internal.amqp.amqpMagic

val serializationProperties: MutableMap<Any, Any> = mutableMapOf()

val testSerializationContext = AMQPSerializationContextImpl(
        deserializationClassLoader = ClassLoader.getSystemClassLoader(),
        whitelist = AllWhitelist,
        properties = serializationProperties,
        objectReferencesEnabled = false,
        useCase = AMQPSerializationContext.UseCase.Testing,
        encoding = null)